src/coreclr/jit/targetarm64.h

```diff
@@ -75,8 +75,15 @@
   #define RBM_FLT_CALLEE_SAVED    (RBM_V8|RBM_V9|RBM_V10|RBM_V11|RBM_V12|RBM_V13|RBM_V14|RBM_V15)
   #define RBM_FLT_CALLEE_TRASH    (RBM_V0|RBM_V1|RBM_V2|RBM_V3|RBM_V4|RBM_V5|RBM_V6|RBM_V7|RBM_V16|RBM_V17|RBM_V18|RBM_V19|RBM_V20|RBM_V21|RBM_V22|RBM_V23|RBM_V24|RBM_V25|RBM_V26|RBM_V27|RBM_V28|RBM_V29|RBM_V30|RBM_V31)
 
+  #define RBM_LOWMASK              (RBM_P0|RBM_P1|RBM_P2|RBM_P3|RBM_P4|RBM_P5|RBM_P6|RBM_P7)
+  #define RBM_HIGHMASK             (RBM_P8|RBM_P9|RBM_P10| RBM_P11|RBM_P12|RBM_P13|RBM_P14|RBM_P15)
+  #define RBM_ALLMASK              (RBM_LOWMASK|RBM_HIGHMASK)
+
+  #define RBM_MSK_CALLEE_SAVED    (0)
+  #define RBM_MSK_CALLEE_TRASH    RBM_ALLMASK
```

Comment:
somewhere I should just zero it out if we are not running on SVE machine.

Reply:
Is the TP cost coming from the additional killed registers? I assume that's because we don't have a predicate registers equivalent of `compFloatingPointUsed`.

I wonder if you could just add a case for predicate registers here:
https://github.com/dotnet/runtime/blob/55f2bc67dd703625cb72fefa4c5a914431639a6c/src/coreclr/jit/lsrabuild.cpp#L3094

And then during allocation, mask out the predicate registers when processing kills if no predicate registers were used.

We would still be creating additional RegRecords though, but maybe this helps a bit.

Reply:
> We would still be creating additional RegRecords though, but maybe this helps a bit.

Actually I guess we were creating those `RegRecord`s even before this PR, so I imagine it would help quite a bit for this PR.

Reply:
I am already doing it in https://github.com/dotnet/runtime/pull/104065/files#diff-ad66a6bcf1fd550d5ad10d995c03218afbbc39463d36e1f2a224f9ca070a2f99R858-R860. Predicate registers exist only in presence of floating point usage. Yes, we do the newly added extra predicate registers in `processKills()` and that's what show up impacting TP. For non-sve arm64 machine, we don't have to iterate through them.

Reply:
> somewhere I should just zero it out if we are not running on SVE machine.

although note that when we altjit, we say that "sve capability enable", so we will see predicate registers and will process them during kills. The TP information will be misleading for those cases, but I will add this anyway so that on non-sve arm64 machine, we do not process them.

Reply:
I think the use of predicate registers is going to be much more rare than using float registers, hence adding this extra check would help regardless.

> I will add this anyway so that on non-sve arm64 machine, we do not process them.

I don't see a good reason to try optimizing for non-SVE machines. In the future we would expect most arm64 machines to be SVE enabled, right?

I think we should rather optimize for the common case of "predicate registers not used". It should be possible now that we are only creating one`RefTypeKill` per call.

Reply:
>I don't see a good reason to try optimizing for non-SVE machines. In the future we would expect most arm64 machines to be SVE enabled, right?

Yes

> I think the use of predicate registers is going to be much more rare than using float registers, hence adding this extra check would help regardless.

Agree. I will do a separate pass for it. https://github.com/dotnet/runtime/issues/104157 to track it.

Reply:
> I don't see a good reason to try optimizing for non-SVE machines

Thinking about this a bit more, I think I will just revert https://github.com/dotnet/runtime/pull/104065/commits/bcfd8a827904ccf0d0ab046fd268cb3c85f243e9 and will do it properly in https://github.com/dotnet/runtime/issues/104157